OpenSim/Examples/Plugins/CoupledBushingForceExample/CoupledBushingForce.cpp

```diff
@@ -89,239 +75,78 @@ CoupledBushingForce::CoupledBushingForce( std::string body1Name, SimTK::Vec3 poi
-    mutableThis->_inb1 = SimTK::Transform(r1, _locationInBody1);
-    mutableThis->_inb2 = SimTK::Transform(r2, _locationInBody2);
-}
-
-//=============================================================================
-// OPERATORS
-//=============================================================================
-//_____________________________________________________________________________
-/**
- * Assignment operator.
- *
- * @return Reference to this object.
- */
-CoupledBushingForce& CoupledBushingForce::operator=(const CoupledBushingForce &aForce)
-{
-    Force::operator=(aForce);
-    copyData(aForce);
-    return(*this);
+/*
+* Construct Structural Connectors
+*/
+void CoupledBushingForce::constructConnectors() {
+    constructConnector<PhysicalFrame>("frame1");
+    constructConnector<PhysicalFrame>("frame2");
 }
 
-//=============================================================================
-// SET
-//=============================================================================
-//_____________________________________________________________________________
-/**
- * Following methods set attributes of the weld Force */
-void CoupledBushingForce::setBody1ByName(std::string aBodyName)
+void CoupledBushingForce::extendFinalizeFromProperties()
 {
-    _body1Name = aBodyName;
-}
+    Super::extendFinalizeFromProperties();
 
-void CoupledBushingForce::setBody2ByName(std::string aBodyName)
-{
-    _body2Name = aBodyName;
-}
+    //mark frames in property list as subcomponents
+    for (int i = 0; i < updProperty_frames().size(); ++i){
+        addComponent(&upd_frames(i));
+    }
 
-/** Set the location and orientation (optional) for weld on body 1*/
-void CoupledBushingForce::setBody1BushingLocation(Vec3 location, Vec3 orientation)
-{
-    _locationInBody1 = location;
-    _orientationInBody1 = orientation;
+    finalizeMatricesFromProperties();
 }
 
-/** Set the location and orientation (optional) for weld on body 2*/
-void CoupledBushingForce::setBody2BushingLocation(Vec3 location, Vec3 orientation)
-{
-    _locationInBody2 = location;
-    _orientationInBody2 = orientation;
-}
 
 //=============================================================================
 // COMPUTATION
 //=============================================================================
 /** Compute the deflection (spatial separation) of the two frames connected
-    by the bushing force. Angualar displacement expressed in Euler angles.
+    by the bushing force. Angular displacement expressed in Euler angles.
     The force and potential energy are determined by the deflection.  */
 SimTK::Vec6 CoupledBushingForce::computeDeflection(const SimTK::State& s) const
 {
-    const Transform& X_GB1 = _b1->getBodyTransform(s);
-
-    const Transform& X_GB2 = _b2->getBodyTransform(s);
+    // get connected frames
+    const PhysicalFrame& frame1 = getConnectee<PhysicalFrame>("frame1");
+    const PhysicalFrame& frame2 = getConnectee<PhysicalFrame>("frame2");
```

Comment:
I will change `Connector::set_connected_to_name()` to `Connector::set_connectee_name()` in a separate PR.